piqe_ocp_lib/api/tasks/operator_ops.py

```diff
@@ -15,68 +17,93 @@
 
 
 class OperatorInstaller(OcpBase):
-    def __init__(self, kube_config_file):
+    def __init__(self, kube_config_file=None):
         super(OperatorInstaller, self).__init__(kube_config_file=kube_config_file)
         self.og_obj = OperatorGroup(kube_config_file=self.kube_config_file)
         self.sub_obj = Subscription(kube_config_file=self.kube_config_file)
         self.ohp_obj = OperatorhubPackages(kube_config_file=self.kube_config_file)
         self.proj_obj = OcpProjects(kube_config_file=self.kube_config_file)
         self.csv = ClusterServiceVersion(self.kube_config_file)
 
-    def _derive_install_mode_from_target_namespaces(self, operator_name, target_namespaces):
+    def _derive_install_mode_from_target_namespaces(self, target_namespaces: Union[list, str]) -> str:
         """
         Length of target_namespaces list enables us to derive the appropriate install mode
         Look at the length of the target_namespaces list:
             1. If it's zero, then we verify if 'AllNamespaces' is supported.
             2. If length is 1, then we verify if 'SingleNamespace' is supported.
             3. If length > 1, then we verify 'MultiNamespace' is supported.
```

Comment:
Can we update this docstring to include the `OwnNamespace` install_mode and mention `AllNamespaces` to require a wildcard rather than zero length?

Reply:
Same as below with `list`, should be `Union[List[str], str]`

Reply:
Will fix